src/compiler/checker.ts

```diff
@@ -6383,6 +6406,36 @@ namespace ts {
                 getUnionType([info1.type, info2.type]), info1.isReadonly || info2.isReadonly);
         }
 
+        function resolveSyntheticJSModuleExportMembers(type: IntersectionType) {
+            if (type.types.length === 2) {

```

Comment:
This is a bad predicate. Intersection types are flattened, so if I have one of these intersections and intersect with a 3rd thing, the members from the js initializer will no longer resolve. (eg, `/* @type {typeof jsContainer & MyThing} */`)

You probably wanna do the filtering/combining on `CheckFlags.SyntheticExportEqual` in the main `resolveIntersectionTypeMembers` block.

Reply:
Woooooops. Nice catch, I forgot all about that. Upon reflection, I decided that removing duplicates from the intersection members wasn't worth that much work anyway. I think it's OK to have duplicated properties of an intersection in a [probably] rare case like this.

I removed `resolveSyntheticJSModuleExportMembers` and updated the baselines.